src/main.rs

```diff
@@ -572,21 +639,27 @@ fn main_recv(
     // time, and if the network is faster the channel will be full all the time.
     let (sender, receiver) = mpsc::sync_channel::<Chunk>(16);
 
+    let total_len: u64 = plan.files.iter().map(|f| f.len).sum();
+
+    let total_bytes_received = Arc::new(parking_lot::Mutex::new(0u64));
+    let start_time = Instant::now();
+    let total_bytes_clone = total_bytes_received.clone();
     let writer_thread = std::thread::spawn::<_, ()>(move || {
-        let total_len: u64 = plan.files.iter().map(|f| f.len).sum();
         let mut files: Vec<_> = plan.files.into_iter().map(FileReceiver::new).collect();
 
-        let start_time = Instant::now();
         let mut bytes_received: u64 = 0;
 
         for chunk in receiver {
             let file = &mut files[chunk.file_id.0 as usize];
-            bytes_received += chunk.data.len() as u64;
+                        let chunk_len = chunk.data.len() as u64;
```

Comment:
This looks indented incorrectly. You can fix it with `rustfmt`, though I’m not sure if we formatted it previously; if `rustfmt` would make changes on master, let’s format it first in a separate pull request, to not pollute the diff here with formatting changes.